Show displayName (translated dashboard title) in edit mode

edoardo opened the pull request:
For the UI to change and reflect the translation update, a new version of d2-ui is required (see: https://github.com/dhis2/d2-ui/pull/186).

edoardo committed:
Show translated value for dashboard name in edit mode

edoardo committed:
Add displayName in the store, used with translations

edoardo committed:
Add support for setting displayName in dashboards/editDashboard

This is used to avoid having to reload the whole dashboard when a
translation is added/edited via the translations dialog in edit mode.

edoardo committed:
Update displayName on successful translation

The change is directly reflected only if the translated string is in the
same language as the current user's language.
The UI in edit mode is then updated when the translation dialog is
closed upon a successful translation.

edoardo committed:
Merge branch 'master' into task/name-displayname-in-edit-mode

jenniferarnesen commented on src/ControlBarContainer/EditBar.js:
```diff
@@ -60,6 +65,25 @@ class EditBar extends Component {
+    onTranslationsSaved = async translations => {
+        if (translations && translations.length) {
+            const uiLocale = await this.context.d2.currentUser.userSettings.get(
+                'keyUiLocale'
```
Are you sure its the UiLocale that is wanted? I thought it would be the dbLocale

edoardo committed:
Simplify displayName handling

Only touch dashboards in Redux store, no need to set the displayName
value also in editDashboard.

edoardo committed:
Remove unused properties in the editDashboard object

edoardo committed:
Merge branch 'master' into task/name-displayname-in-edit-mode

edoardo committed:
Rename variable
